src/TerminalLogger.jl

```diff
@@ -342,6 +344,9 @@ function handle_message(logger::TerminalLogger, level, message, _module, group,
             pop!(logger.sticky_messages, id)
         end
     else
+        for _ in 1:logger.margin
+            println(logger.stream)
```

Comment:
Let's move this up to print it into `iob` instead of directly to logger.stream.  This helps make logging to the underlying stream atomic (preventing garbled messages with `@async`)

Reply:
Okay! Take a look at my latest commit